### DISCUSSION TITLE
Passing props breaking the code

### DISCUSSION THREAD

**Discussion by lilyntalmers** [ivan-dalmet/formiz#61] [Q&A] [answered]
I have a prop(`myurl={baseUrl}`) in `App.js` file. And I am trying to pass it to my Formiz form `MyForm.js`. Inside `MyForm.js` I have added `MyUrl={this.props.myurl}` on line 82. But the form is not working.

[CodeSandbox.io](https://codesandbox.io/s/formiz-pricing-wizard-forked-2k7u4?file=/src/MyForm.js:2184-2208)

I strongly believe I have done something wrong but can't figure out what😅. Would really appreciate if somebody could take a look.

**Comment by yoannfleurydev** [comment, marked as answer]
Hi @lilyntalmers, 👋

you forgot to put `props` as a `MyForm` parameter:

```jsx
const MyForm = (props) => { // 👈 here
  const myForm = useForm();
```  

also, when using it later, you don't need `this` as you are in a functional component

```jsx
<Step1
  form={myForm}
  totalPrice={totalPrice}
  myUrl={props.myurl}
/>
```

You can also [destructure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) props to get only the `myurl` variable:

```jsx
// at the beginning of the component
const { myurl } = props;
// ...
// and then, later on in the return
 <Step1
  form={myForm}
  totalPrice={totalPrice}
  myUrl={myurl}
/>
```

or directly in the component parameter

```jsx
const MyForm = ({ myurl }) => { // 👈 here
  const myForm = useForm();
```  

Quick note: avoid the prop `MyUrl` and prefer `myUrl` as **camelCase** is the convention, and not *PascalCase*.

I hope I helped, please let me know if it is working for you.
Have a nice day. 😊

**Reply by lilyntalmers** [reply]
It totally worked. Thank you very much @yoannfleurydev.

I used this
```
// at the beginning of the component
const { myurl } = props;
// ...
// and then, later on in the return
 <Step1
  form={myForm}
  totalPrice={totalPrice}
  myUrl={myurl}
/>
```
Also I had to pass it here:
`const data = { ...values, totalPrice, myurl };`

Hope you have a nice day too. 😊
